mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp

```diff
@@ -264,13 +299,16 @@ void createGemmTuningRangeBF(TuningParamSet *newSpace,
                                                gemmKPerBlock, gemmMPerWave,
                                                gemmMnPerXdl, gemmKPack,
                                                splitKFactor, forceUnroll, true);
+                    bool nonExhaustiveSkipCondition =
+                        canSkipConfigForReductionFusion(gemmOp, gemmMPerBlock,
+                                                        gemmNPerBlock) ||
+                        failed(tuningInfo.couldBePerformant(info, gemmParams));
                     if (gemmMPerBlock >= gemmMPerWave &&
                         gemmNPerBlock >= gemmMnPerXdl) {
                       if (succeeded(tuningInfo.paramsProbablyValid(
                               b, info, gemmParams)) &&
                           (kind == TuningParamSetKind::Exhaustive ||
-                           succeeded(
-                               tuningInfo.couldBePerformant(info, gemmParams))))
+                           !nonExhaustiveSkipCondition))
```

Comment:
Could you implement this into `couldBePerformant()` please? Maybe extend that function to take a `bool isReduction`?

Reply:
... This is also, IMO, adding a `has_reduction` to the problem key, so that MIGraphX doesn't cache reduction and non-reduction GEMMs together

Reply:
I moved into couldBePerformant() and I d defer problem key change to be in a different PR for bisectability.